### ISSUE TITLE
Federated status

### ISSUE THREAD

**Issue by gabek** [owncast/owncast#2912]
### Share your bug report, feature request, or comment.

This would be a two part addition:

- Piggyback on the existing "Go live" ActivityPub activity that gets sent to followers with additional metadata (TBD what that is or where) that shares via presence status indication that the stream is now live. This additional metadata would be primarily for other Owncast servers to show in the "mini directory" when a stream is live.
- Send an additional ActivityPub activity when a stream ends so the other Owncast servers that are displaying status will know it's now offline.

Best case would be to work with other ActivityPub powered services that may either be right now using presence status or plan to in the future, so we can share a standard and other interfaces would be able to show when a stream is live.

**Comment by gabek** [comment]
Was talking to Will from [Immers Space](https://web.immers.space/) and they use the `Arrive` and `Leave` activities for federating presence. This would make sense to do the same for interop. They use a [`Place` object](https://www.w3.org/TR/activitystreams-vocabulary/#dfn-place) with the `url` property for the `location` property of the activity. 
https://www.w3.org/TR/activitystreams-vocabulary/#dfn-arrive

<blockquote><div><strong><a href="https://www.w3.org/TR/activitystreams-vocabulary/#dfn-arrive">Activity Vocabulary</a></strong></div></blockquote>

**Comment by gabek** [comment]
In retrospect we will not be able to piggyback on the existing "Go live" for sending out this presence since you cannot send multiple ActivityPub activities in a payload (that I'm able to see, anyway). And even if we were sending out multiple objects within the same activity, Mastodon's handling of multiple objects within a single activity [has been broken](https://github.com/mastodon/mastodon/issues/19111) and just throws the entire payload away.

**Comment by ShadowJonathan** [comment]
Does this feature include the ability to re-stream via one's own instance, or will the user just be redirected to the remote stream?

(Maybe the ability to authenticate the user via the local instance automatically in the remote chat? Or maybe this is scope creep, and should be spun out into other issues)

**Comment by gabek** [comment]
I don't completely understand the question, regarding re-streaming.

The functionality here would be to simply federate the online status (online/offline) of an Owncast stream. It has nothing to do with the video itself.

**Comment by ShadowJonathan** [comment]
Right, but could it be possible for users to view the remote stream via the "local" instance? (maybe via an embedded video player)

Plus the possibility to automatically authenticate the local user in the remote chat as coming from the local instance, when clicking to watch the stream.

**Comment by gabek** [comment]
I think all of that would fall under different functionality. All this issue is for is the ActivityPub implementation for federated status. I think what you're talking about might be more under the banner of "raids"?  If so, https://github.com/owncast/owncast/discussions/717 might be more of the place to discuss that.

<blockquote><img src="https://opengraph.githubassets.com/b165619b41565929d73d467b4bca0e55c2488bc88b125edcbb19aad9847701f7/owncast/owncast/discussions/717" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/owncast/owncast/discussions/717">Raids: How should they work with Owncast? · owncast/owncast · Discussion #717</a></strong></div><div>For those unfamiliar, on Twitch a "Raid" is when a streamer ends their broadcast and sends all of their viewers to another stream. To have something with a similar feel, but in a decentralized mann...</div></blockquote>

**Comment by ShadowJonathan** [comment]
not really

I mean something more akin to this scenario; a user logs into their local owncast instance, and sees which of their (followed) friends have gone live. Then, they click on one of them, and they get their stream loaded in the website of their local owncast instance, and the remote chat loaded automatically, authenticated as the user of their local instance

**Comment by gabek** [comment]
Take a look at https://github.com/owncast/owncast/discussions/717#discussioncomment-364678, I think this is what you're talking about.

But if you have different specifics, please open another issue, as this issue is just about ActivityPub implementation, not user-facing features.

<blockquote><img src="https://opengraph.githubassets.com/b165619b41565929d73d467b4bca0e55c2488bc88b125edcbb19aad9847701f7/owncast/owncast/discussions/717" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/owncast/owncast/discussions/717">Raids: How should they work with Owncast? · owncast/owncast · Discussion #717</a></strong></div><div>For those unfamiliar, on Twitch a "Raid" is when a streamer ends their broadcast and sends all of their viewers to another stream. To have something with a similar feel, but in a decentralized mann...</div></blockquote>

**Comment by ShadowJonathan** [comment]
Did this get implemented? 👀

**Comment by gabek** [comment]
Yes! It'll be released in `v0.3.0`. We're discussing it in chat if you want to drop by and talk about it. https://matrix.to/#/#owncast.community:matrix.org

<blockquote><div><strong><a href="https://matrix.to/#/#owncast.community:matrix.org">You're invited to talk on Matrix</a></strong></div><div>You're invited to talk on Matrix</div></blockquote>
